src/Tools/TestDiscoveryWorker/Program.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading.Channels;
+
+using Xunit;
+using Xunit.Abstractions;
+
+if (args.Length <= 0)
+{
+    return;
+}
+
+using var pipeClient = new AnonymousPipeClientStream(PipeDirection.In, args[0]);
+using var sr = new StreamReader(pipeClient);
+// Display the read text to the console
+string? output;
+
+// Wait for 'sync message' from the server.
+do
+{
+    output = sr.ReadLine();
+}
+while (!(output?.StartsWith("ASSEMBLY") == true));
+
+if ((output = sr.ReadLine()) is not null)
+{
+    var assemblyFileName = output;
+
+#if NET6_0_OR_GREATER   
+    var resolver = new System.Runtime.Loader.AssemblyDependencyResolver(assemblyFileName);
+    System.Runtime.Loader.AssemblyLoadContext.Default.Resolving += (context, assemblyName) =>
+    {
+        var assemblyPath = resolver.ResolveAssemblyToPath(assemblyName);
+        if (assemblyPath != null)
+        {
+            return Assembly.LoadFrom(assemblyPath);
```

Comment:
```suggestion
            return context.LoadFromAssemblyPath(assemblyPath);
```

Yes the code as written is equivalent because `Assembly.LoadFrom` will go into the default context, but think this code is more portable.